.github/workflows/docker_build.yml

```diff
@@ -47,6 +48,16 @@ jobs:
         with:
           fetch-depth: "0"
 
+      - name: 'use component inputs as component var'
+        if: ${{ inputs.component != ''}}
+        run: echo "COMPONENT_NAME=${{ env.component }}" >> $GITHUB_ENV
+
+      - name: 'set component var in case someone forget to set it'
+        if: ${{ inputs.component == ''}}
+        # 'remove owner/ from reponame'
+        run: |
+          echo "COMPONENT_NAME=${GITHUB_REPOSITORY##$GITHUB_REPOSITORY_OWNER/}" >> $GITHUB_ENV
+
```

Comment:
maybe a simpler way, I didn't test it though

```yaml
      - name: 'set component with repo name'
        env:
          COMPONENT_NAME: ${{ github.event.repository.name }}

      - name: 'use component inputs as component var'
        if: ${{ inputs.component != ''}}
        env:
           COMPONENT_NAME: ${{ inputs.component }}
```

Reply:
seems pretty all webhook payloads have `repository` field so that's better :D

Reply:
any reason to not use `env:` but  `run: echo "....." >> $GITHUB_ENV`, the `env:` looks more native&clear way

Reply:
do it support dynamic definition and logic or operator?

```yaml
jobs:
  build:
    env: ${{ inputs.component || github.event.repository.name }}
```

Reply:
looks like it actually works. a lesson learned 👍